### ISSUE TITLE
When downloading the pnb files the extension is xml

### ISSUE THREAD

**Issue by phebal** [galterlibrary/digital-repository#490]
Should be pdf obviously, script was bugged.

**Comment by phebal** [comment]
Wow, so this is almost impossible. There is no way to update file datastream attributes (metadata) in ActiveFedora without also updating the content.  The only AF API-based way is to create a new version of file and re-send the content.

GenericFile#content.ldp_source.update almost does the right thing but it posts a blank body along with the headers so it's not an option. At this point I think the only sensible way is Fedora's own API.

Gha...

**Comment by vioil** [comment]
Wow, I didn't notice that

**Comment by phebal** [comment]
The browser opens them just fine, however when you try to save them the problem manifest itself.

**Comment by phebal** [comment]
`PATCH`:
http://127.0.0.1:8983/fedora/rest/dev/05/74/1r/77/05741r77z/content/fcr:metadata

Where the URI can be generated with:
```ruby
GenericFile.first.content.ldp_source.subject + "/fcr:metadata"
```
Headers:
`Content-Type: application/sparql-update`

In body: SPARQL query to replace the filename:
```sparql
PREFIX ebucore: <http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#>
DELETE {
  <> ebucore:filename "<old_filename>" .
}
INSERT {
  <> ebucore:filename "<new_filename>" .
}
WHERE { }
```

File name can be determined with:
```ruby
GenericFile.first.content.original_name
```

**Comment by phebal** [comment]
Wow, finally fixed after completing #491

**Comment by vioil** [comment]
awesome
